web3/_utils/request.py

```diff
@@ -86,7 +94,8 @@ def get_response_from_get_request(
     endpoint_uri: URI, *args: Any, **kwargs: Any
 ) -> requests.Response:
     kwargs.setdefault("timeout", DEFAULT_TIMEOUT)
-    session = cache_and_return_session(endpoint_uri)
+    provider_id = kwargs.get("provider_id")
```

Comment:
We should update all the async versions of these as well and add a similar integration test for the `AsyncHTTPProvider` to make sure it passes.